### PR TITLE
Add chunking and PSR-7 support

### PR DESCRIPTION
**Pull request by jasonhebert**
Add support for parsing MessageInterface and StreamInterface objects
Add support for chunking input for processing

### CODE REVIEW

**Comment by MAXakaWIZARD** [comment]
Could you please add some unit tests for this feature?

**Comment by jasonhebert** [comment]
I can add additional testing, no problem. Update coming shortly.

**Comment by jasonhebert** [comment]
Additional tests added, has coverage increased to an acceptable level? I think I am testing all available items. I think the code coverage is balking at the StreamWrapper class. I'll see if I can further test those methods.

EDIT: I see where the testing lacks now and am on it...

**Comment by jasonhebert** [comment]
Test coverage now at 100%

**Comment by jasonhebert** [comment]
Any thoughts on a time-frame for merging this? Thx...

**Comment by MAXakaWIZARD** [comment]
@jasonhebert sorry for delay, will review today

File: src/Parser.php [hunk 1]
```diff
@@ -35,107 +38,67 @@ public function __construct()
...
+     * @param int               $size         Size of the chunk to collect before processing
+     * @param string|resource   $input        File path or resource
      * @param callback|callable $itemCallback Callback
+     * @param bool              $assoc        Parse as associative arrays
      *
-     * @throws \Exception
+     * @throws Exception
      */
-    public function parseAsObjects($input, $itemCallback)
+    public function chunk(int $size, $input, $itemCallback, bool $assoc = true): void
```
Comment (MAXakaWIZARD) [inline src/Parser.php]: @jasonhebert any strong argument why are you putting $size as the first param? Looks like not very consistent with ->parse method.

Comment (jasonhebert) [reply src/Parser.php]: none, will modify

Comment (jasonhebert) [reply src/Parser.php]: Corrected in latest commit

**Comment by MAXakaWIZARD** [comment]
@jasonhebert thanks for contribution! I will release new version soon

**Pull request merged**